src/openrct2/peep/Guest.cpp

```diff
@@ -2298,7 +2298,7 @@ bool Guest::ShouldGoToShop(Ride* ride, bool peepAtShop)
     }
 
     // Basic price checks
-    if (ride->price != 0 && ride->price > cash_in_pocket)
+    if (ride_get_price(ride) != 0 && ride_get_price(ride) > cash_in_pocket)
```

Comment:
Perhaps better to save the result in a variable, rather than calling this twice.